docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
+
+
+### Current BYOH release: `v0.3.0`
+
+---
+##### Follow below steps to make a new release in BYOH
+1. Update `metadata.yaml` file if a new major/minor version is being released. This is not needed in case of patch release. Map the newly added version to supported ClusterAPI contract version.
+
+2. Create new tag from main branch. For example - `git tag v0.3.0`
+
+3. Push the tag to vmware-tanzu/cluster-api-provider-byoh repository (not in custom fork)
+`git push v0.3.0`
+
+4. This will trigger a new workflow in GitHub actions. Wait for the workflow to finish. This workflow will create a draft release having the contributions, changes and assets. Verify the assets, they should contain BYOH agent, cluster templates and infrastructure components.
+
+5. Next step is to release BYOH controller image.  Create controller image in local using below command
```

Comment:
```suggestion
5. Next step is to release `BYOH` controller OCI image.  Create controller image in local using below make target
```

Reply:
Add a note that this is a manual step and TODO: Automate this step